File: 500/txtindex.mjs
import fs from 'fs/promises';
import path from 'path';

const indexFilePath = './index-txt.json';
const sourceDir = 'c:/users/owner/downloads/listing/pages';
const destinationDir = 'c:/users/owner/downloads/listing/500html';

async function moveFiles() {
  try {
    // Read and parse the index file
    const indexContent = await fs.readFile(indexFilePath, 'utf-8');
    const fileIndex = JSON.parse(indexContent);

    // Ensure the destination directory exists
    await fs.mkdir(destinationDir, { recursive: true });

    let movedCount = 0;

    // Iterate over files in the index
    for (const [fileName, isTrue] of Object.entries(fileIndex)) {
      if (isTrue) {
        const sourcePath = path.join(sourceDir, fileName);
        const destinationPath = path.join(destinationDir, fileName);

        try {
          // Move the file
          await fs.rename(sourcePath, destinationPath);
          console.log(`Moved: ${fileName}`);
          movedCount++;
        } catch (err) {
          if (err.code === 'ENOENT') {
            console.error(`File not found: ${fileName}`);
          } else {
            console.error(`Error moving file: ${fileName}`, err);
          }
        }
      }
    }

    console.log(`\nTotal files moved: ${movedCount}`);
  } catch (err) {
    console.error('Error reading index or processing files:', err);
  }
}

moveFiles();

File: nlp-py/txtindex.py
import os
import shutil
import json

# Paths
index_file = r"./index.json"
source_folder = r"c:\users\owner\downloads\listing\pages"
destination_folder = r"c:\users\owner\downloads\listing\500html"

# Ensure the destination folder exists
os.makedirs(destination_folder, exist_ok=True)

# Load the index.json
with open(index_file, 'r', encoding='utf-8') as f:
    index = json.load(f)

# Move files containing the keyword 'javascript'
moved_files_count = 0
total_files = len(index)
processed_files_count = 0

for filename, metadata in index.items():
    processed_files_count += 1
    print(f"Processing {processed_files_count} of {total_files}: {filename}")
    
    # Check if the file has the keyword "javascript"
    if any("javascript" in str(value).lower() for value in metadata.values()):
        source_path = os.path.join(source_folder, filename)
        destination_path = os.path.join(destination_folder, filename)

        # Move the file if it exists in the source folder
        if os.path.exists(source_path):
            shutil.move(source_path, destination_path)
            print(f"Moved: {filename}")
            moved_files_count += 1
        else:
            print(f"File not found, skipping: {filename}")

print(f"Total files moved: {moved_files_count}")
